test/e2e/node/pods.go

```diff
@@ -857,6 +857,183 @@ var _ = SIGDescribe("Pod Extended (container restart policy)", framework.WithFea
 	})
 })
 
+var _ = SIGDescribe("Pod Extended (RestartAllContainers)", framework.WithFeatureGate(features.ContainerRestartRules), framework.WithFeatureGate(features.RestartAllContainersOnContainerExits), func() {
```

Comment:
Can you add similar tests for StandaloneMode kubelet with static pods? It is important to check that this is not introducing unnecessary dependencies on API server